src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -698,7 +698,15 @@ private static IpcMessage CreateWriteDumpMessage(DumpType dumpType, string dumpP
                 throw new ArgumentNullException($"{nameof(dumpPath)} required");
             }
 
-            byte[] payload = SerializePayload(dumpPath, (uint)dumpType, logDumpGeneration);
+            // Quote the path to handle spaces correctly in createdump on Windows only
+            // Only add quotes if the path is not already quoted
+            // This is only needed on Windows where the runtime builds the command line for createdump
+            string pathToUse = dumpPath;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Does this cause an issue if the tool is running on Windows but it is connecting to a non-Windows runtime via a proxy like dsrouter? The check only tells us the OS running the tool, not the OS that will run createdump.

The [ProcessInfo command](https://github.com/dotnet/diagnostics/blob/main/documentation/design-docs/ipc-protocol.md#processinfo) reports the OS being used by the runtime.